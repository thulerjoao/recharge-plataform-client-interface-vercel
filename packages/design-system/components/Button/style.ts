import styled, { keyframes } from "styled-components";
import { Theme } from "../../theme/theme";

interface ButtonInputProps {
  height: 28 | 32 | 40 | 48;
  rounded: boolean;
  loading: boolean;
  shadow: boolean;
  margin: string;
}

export const ButtonInput = styled.button<ButtonInputProps>`
  user-select: none;
  font-size: ${({ height }) => (height >= 40 ? "16px;" : "14px;")};
  font-weight: 600;
  width: 100%;
  height: ${({ height }) => `${height}px`};
  align-items: center;
  background-color: ${Theme.colors.mainhighlight};
  border: none;
  box-shadow: ${({ shadow }) =>
    shadow && `0px 0px 12px 0px ${Theme.colors.mainhighlight}`};
  border-radius: ${({ height }) => (height >= 40 ? "16px;" : "8px;")};
  border-radius: ${(props) => props.rounded && `${props.height / 2}px`};
  display: flex;
  justify-content: center;
  text-align: center;
  ${(props) => !props.loading && !props.disabled && "cursor: pointer;"}
  ${({ margin }) => `margin: ${margin};`}
`;

const spin = keyframes`
  to {
    transform: rotate(360deg);
  }
`;

export const Spinner = styled.div`
  border: 2px solid #071116;
  border-top-color: transparent;
  border-radius: 50%;
  width: 18px;
  height: 18px;
  margin-left: 0px;
  animation: ${spin} 0.8s linear infinite;
`;
