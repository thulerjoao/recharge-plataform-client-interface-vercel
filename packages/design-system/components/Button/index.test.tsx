import { render } from "@testing-library/react";
import Button from "./index";

describe("Button Component Snapshots", () => {
  const heights = [28, 32, 40, 48] as const;

  heights.forEach((height) => {
    test(`matches snapshot with height ${height}`, () => {
      const { asFragment } = render(
        <Button title={`Height ${height}`} height={height} />,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  test("matches snapshot with rounded corners", () => {
    const { asFragment } = render(
      <Button title="Rounded Button" height={40} rounded />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches snapshot with no rounded corners", () => {
    const { asFragment } = render(
      <Button title="Square Button" height={40} rounded={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
