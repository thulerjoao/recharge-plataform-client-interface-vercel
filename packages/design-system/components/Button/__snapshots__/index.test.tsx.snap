// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshots matches snapshot with height 28 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 14px;
  font-weight: 600;
  width: 100%;
  height: 28px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Height 28
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshots matches snapshot with height 32 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 14px;
  font-weight: 600;
  width: 100%;
  height: 32px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Height 32
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshots matches snapshot with height 40 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 16px;
  font-weight: 600;
  width: 100%;
  height: 40px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 16px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Height 40
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshots matches snapshot with height 48 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 16px;
  font-weight: 600;
  width: 100%;
  height: 48px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 16px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Height 48
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshots matches snapshot with no rounded corners 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 16px;
  font-weight: 600;
  width: 100%;
  height: 40px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 16px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Square Button
    </p>
  </button>
</DocumentFragment>
`;

exports[`Button Component Snapshots matches snapshot with rounded corners 1`] = `
<DocumentFragment>
  .c0 {
  user-select: none;
  font-size: 16px;
  font-weight: 600;
  width: 100%;
  height: 40px;
  align-items: center;
  background-color: #00C8FF;
  border: none;
  border-radius: 16px;
  border-radius: 20px;
  display: flex;
  justify-content: center;
  text-align: center;
  cursor: pointer;
  margin: undefined;
}

<button
    class="c0"
  >
    <p>
      Rounded Button
    </p>
  </button>
</DocumentFragment>
`;
